modules/apps/portal/portal-messaging/src/main/java/com/liferay/portal/messaging/internal/DefaultMessageBus.java

```diff
@@ -478,19 +351,178 @@ private void _updateDestination(
 	private static final Log _log = LogFactoryUtil.getLog(
 		DefaultMessageBus.class);
 
+	private BundleContext _bundleContext;
+	private ServiceTracker<DestinationEventListener, DestinationEventListener>
+		_destinationEventListenerServiceTracker;
 	private final Map<String, Destination> _destinations =
```

Comment:
@W2jx As you already have service tracker for destinations, you should not need this map here. Please use service tracker map for destination and remove it here.